### PR TITLE
Review the bridge and fix minor bugs

### PR DESCRIPTION
**Pull request by Lbqds**

### CODE REVIEW

File: node/pkg/alephium/watcher.go [hunk 1]
```diff
@@ -236,14 +232,13 @@ func (w *Watcher) fetchEvents(ctx context.Context, logger *zap.Logger, client *C
 						errC <- err
 						return
 					}
+					logger.Info("received a message", zap.String("txId", unconfirmed.TxId), zap.String("blockHash", unconfirmed.BlockHash))
 					if unconfirmed.msg.IsAttestTokenVAA() {
-						logger.Info("received an attest token message", zap.String("txId", unconfirmed.TxId), zap.String("blockHash", unconfirmed.BlockHash))
```
Comment (h0ngcha0) [inline node/pkg/alephium/watcher.go:240]: Q: Does it make sense to add a `zap.String("type", "attest")` or `zap.String("type", "transfer")` in log to distinguish the log entry as before? maybe it helps with debugging?

**Review by h0ngcha0 — COMMENTED** [review]
Great work :+1:

File: node/pkg/alephium/watcher.go [hunk 2]
```diff
@@ -396,6 +384,10 @@ func (w *Watcher) handleEvents_(
 					continue
 				}
 
+				if !*isCanonical {
```
[the remark this answers is not in the document]
Comment (Lbqds) [reply node/pkg/alephium/watcher.go:389]: Yes, otherwise we might lose some events.

**Review by h0ngcha0 — APPROVED** [review]
LGTM :+1:

File: alephium/contracts/token_bridge/attest_token_handler.ral [hunk 3]
```diff
@@ -2,15 +2,15 @@ Contract AttestTokenHandler(
...
+  targetTokenBridgeId: ByteVec,
   mut receivedSequence: U256,
   isLocalHandler: Bool
 ) extends Constants(), TokenBridgeConstants() {
   @using(updateFields = true)
   fn parseAttestToken(vaa: ByteVec) -> (ByteVec, ByteVec, ByteVec, U256, U256) {
     let (emitterChainId, vaaTargetChainId, emitterAddress, msgSequence, payload) = governance.parseAndVerifyVAA(vaa, false)
     assert!(emitterChainId == targetChainId, ErrorCodes.InvalidEmitChainId)
-    assert!(emitterAddress == tokenId!(targetTokenBridge), ErrorCodes.InvalidEmitAddress)
+    assert!(emitterAddress == targetTokenBridgeId, ErrorCodes.InvalidEmitAddress)
```
Comment (polarker) [inline alephium/contracts/token_bridge/attest_token_handler.ral:13]: Q: the emitter address is contract id, not address?

File: node/cmd/guardiand/node.go [hunk 4]
```diff
@@ -219,7 +219,7 @@ func init() {
 	alphRPC = NodeCmd.Flags().String("alphRPC", "", "Alephium RPC URL (required)")
 	alphApiKey = NodeCmd.Flags().String("alphApiKey", "", "Alphium RPC api key")
 	alphMinConfirmations = NodeCmd.Flags().Uint8("alphMinConfirmations", 1, "The min confirmations for alephium tx")
-	alphFetchPeriod = NodeCmd.Flags().Uint8("alphFetchPeriod", 10, "The fetch events period for alwphium watcher")
+	alphPollInterval = NodeCmd.Flags().Uint8("alphPollInterval", 10, "The fetch events period for alwphium watcher")
```
Comment (polarker) [inline node/cmd/guardiand/node.go:222]: Does this mean we will fetch events every 10 seconds in production?

File: alephium/contracts/token_bridge/attest_token_handler.ral (hunk 3 above)
Comment (Lbqds) [reply alephium/contracts/token_bridge/attest_token_handler.ral:13]: This is a naming convention in wormhole, where all emitters on the chain are named `emitterAddress`. So I have kept this naming. And in Alephium, the emitter address is the contract id.

File: node/cmd/guardiand/node.go (hunk 4 above)
Comment (Lbqds) [reply node/cmd/guardiand/node.go:222]: No, it's just a default value, We can specify different values based on different networks.

Comment (polarker) [reply node/cmd/guardiand/node.go:222]: Let's make the default value `4 seconds`

**Review by polarker — APPROVED** [review]
LGTM

**Pull request merged**